### PR TITLE
Add footer template and pattern

### LINKED ISSUE #27: Update footer template part
Update the footer template part so that it matches the [desktop](https://www.figma.com/file/nf4oV51c1JBxJKXshMe7cd/New-Sensei-Theme?node-id=20020070%3A2923) and [mobile](https://www.figma.com/file/nf4oV51c1JBxJKXshMe7cd/New-Sensei-Theme?node-id=20020113%3A1706) designs. Example PR [here](https://github.com/Automattic/themes/pull/6488) and [here](https://github.com/Automattic/themes/pull/6564).

In the screenshots below:
- We should use the navigation block for all the links, even the social ones.
- For the social links, we can link to `#`. The user can replace them with valid URLs.
- "A Really Long Site Title" and "Cours" are the site title.
- "Course Theme" should not be a link. Instead, "Sensei" should be a link.

## Desktop
Screen Shot 2022-10-11 at 3 11 30 PM

## Mobile
Screen Shot 2022-10-11 at 3 10 56 PM

#### Scenario: Footer
**Given** The Sensei theme is activated
**When** I view a page
**Then** the footer looks similar to the design on both desktop and mobile
**And** clicking on a page should take you to the URL that the user configured for that page
**And** clicking on a social network should take you to the URL that the user configured for that social network
**And** clicking on Sensei goes to https://senseilms.com/
**And** clicking on WordPress goes to https://wordpress.org.

### PR DESCRIPTION
**Pull request by Imran92**
Implements https://github.com/Automattic/sensei-theme/issues/27

## Changes proposed

Added pattern and template for footer with dynamic site title font size and URLs

**Note:** The dynamic font size is implemented only for the frontend.
## Testing instructions

- Go to any page in the frontend
- Make sure the footer looks like the design
- Try with different site titles
- Make sure the site title showing part in the bottom holds up looks alright in different screen sizes

## Screenshot

**Desktop**
Screen Shot 2022-11-03 at 10 39 05 PM

**Desktop with long title**
Screen Shot 2022-11-03 at 10 41 08 PM

**Mobile**
Screen Shot 2022-11-03 at 10 39 57 PM

### CODE REVIEW

File: patterns/footer.php [hunk 1]
```diff
@@ -7,81 +7,80 @@
...
+<h3 class="has-gothic-font-family has-medium-font-size" style="line-height:1;text-transform:uppercase">social</h3>
+<!-- /wp:heading --></div>
+<!-- /wp:column -->
 
-<!-- wp:paragraph -->
-<p>Video</p>
+<!-- wp:column {"verticalAlignment":"center","width":"50%","layout":{"type":"default"}} -->
+<div class="wp-block-column is-vertically-aligned-center" style="flex-basis:50%"><!-- wp:group {"layout":{"type":"flex","flexWrap":"wrap"},"fontFamily":"eb-garamond"} -->
+<div class="wp-block-group has-eb-garamond-font-family"><!-- wp:paragraph {"fontSize":"small"} -->
+<p class="has-small-font-size"><a href="#">Instagram</a></p>
```
Comment (donnapep) [inline patterns/footer.php:23]: We need to ensure we're translating all text.

File: patterns/footer.php [hunk 2]
```diff
@@ -7,81 +7,80 @@
...
+<div class="wp-block-columns"><!-- wp:column {"width":"81.7%","style":{"typography":{"lineHeight":"1.3"}}} -->
+<div class="wp-block-column" style="line-height:1.3;flex-basis:81.7%"><!-- wp:columns {"style":{"border":{"left":{"width":"1px"}},"spacing":{"padding":{"left":"20px"},"blockGap":{"top":"10px","left":"20px"}}},"className":"course-footer-links-vertical-space"} -->
+<div class="wp-block-columns course-footer-links-vertical-space" style="border-left-width:1px;padding-left:20px"><!-- wp:column {"verticalAlignment":"center","width":"148px"} -->
+<div class="wp-block-column is-vertically-aligned-center" style="flex-basis:148px"><!-- wp:heading {"level":3,"style":{"typography":{"textTransform":"uppercase","lineHeight":"1"}},"className":"has-gothic-font-family","fontSize":"medium"} -->
+<h3 class="has-gothic-font-family has-medium-font-size" style="line-height:1;text-transform:uppercase">social</h3>
```
Comment (donnapep) [inline patterns/footer.php]: We need to ensure we're internationalizing all strings so they can be translated.

Comment (Imran92) [reply patterns/footer.php]: Thanks @donnapep, updated [here](https://github.com/Automattic/sensei-theme/pull/70/commits/15b53b9b486c757e63bc948236d7f1f16ff69bae)

Comment (LavaGolem) [reply patterns/footer.php]: @donnapep I reviewed the changes requested and it looks good so I'm gonna approve

File: patterns/footer.php [hunk 3]
```diff
@@ -80,7 +80,26 @@
 
 <!-- wp:group {"align":"full","style":{"border":{"top":{"width":"1px"}}},"backgroundColor":"background","className":"course-negative-space-footer","layout":{"type":"constrained","contentSize":""}} -->
 <div class="wp-block-group alignfull course-negative-space-footer has-background-background-color has-background" style="border-top-width:1px"><!-- wp:paragraph {"align":"center","style":{"typography":{"letterSpacing":"0.02em"},"spacing":{"padding":{"top":"80px","bottom":"80px"}}},"backgroundColor":"background","fontSize":"xx-small","fontFamily":"system"} -->
-<p class="has-text-align-center has-background-background-color has-background has-system-font-family has-xx-small-font-size" style="padding-top:80px;padding-bottom:80px;letter-spacing:0.02em"><span style="text-decoration: underline;">Course Theme</span> by <a href="https://senseilms.com/">Sensei</a>, Powered by <span style="text-decoration: underline;">WordPress</span>.</p>
```
Comment (LavaGolem) [inline patterns/footer.php]: @Imran92  is this supposed to be translated or this note we leave untranslated ? 

>Course Theme by [Sensei](https://senseilms.com/), Powered by [WordPress](https://wordpress.org/).

Comment (Imran92) [reply patterns/footer.php]: Based on [this](https://github.com/Automattic/themes/blob/696e9387df923b9fec84dbf2ad513e887822b176/pixl/patterns/footer.php#L23-L25), I think we should probably translate this one too

**Review by LavaGolem — DISMISSED** [review]

File: patterns/footer.php (hunk 2 above)
Comment (LavaGolem) [reply patterns/footer.php]: @donnapep  I've checked the changes requested and it all looks good so I'm gonna approve it

**Review by LavaGolem — APPROVED** [review]

File: patterns/footer.php (hunk 2 above)
Comment (donnapep) [reply patterns/footer.php]: @LavaGolem I didn't do a full review of this, just noticed the strings. Did you review the entire PR?

**Pull request merged**